lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -290,3 +297,363 @@ class CourseGradingPolicy(CourseViewMixin, ListAPIView):
+
+          * responsive_ui: (boolean) Whether or not the block's rendering obtained via block_url is responsive.
+            Returned only if "responsive_ui" is included in the "fields" parameter.
+
+        * navigation: A dictionary that maps block IDs to a collection of navigation information about each block.
+          Each block contains the following fields.
+
+          * descendants: (list) A list of IDs of the children of the block if the block's depth in the
+            course hierarchy is less than the navigation_depth.  Otherwise, a list of IDs of the aggregate descendants
+            of the block.
+
+        * blocks_navigation: A dictionary that combines both the blocks and navigation data.
+
+    """
+    class RequestInfo(object):
+        """
+        A class for encapsulating the request information, including what optional fields are requested.
+        """
+        DEFAULT_FIELDS = "children,graded,format,responsive_ui"
+
+        def __init__(self, request, course):
+            self.request = request
+            self.course = course
+            self.field_data_cache = None
+
+            # check what fields are requested
+            try:
+                # fields
+                self.fields = set(request.GET.get('fields', self.DEFAULT_FIELDS).split(","))
+
+                # children
+                self.children = 'children' in self.fields
+                self.fields.discard('children')
+
+                # block_count
+                self.block_count = request.GET.get('block_count', "")
+                self.block_count = (
+                    self.block_count.split(",") if self.block_count else []
+                )
+
+                # navigation_depth
+                self.navigation_depth = int(request.GET.get('navigation_depth', '3'))
+
+                # block_json
+                self.block_json = json.loads(request.GET.get('block_json', "{}"))
+                if self.block_json and not isinstance(self.block_json, dict):
+                    raise ParseError
+            except:
+                raise ParseError
+
+    class ResultData(object):
+        """
+        A class for encapsulating the result information, specifically the blocks and navigation data.
+        """
+        def __init__(self, return_blocks, return_nav):
+            self.blocks = {}
+            self.navigation = {}
+            if return_blocks and return_nav:
+                self.navigation = self.blocks
+
+        def update_response(self, response, return_blocks, return_nav):
+            """
+            Updates the response object with result information.
+            """
+            if return_blocks and return_nav:
+                response["blocks+navigation"] = self.blocks
+            elif return_blocks:
+                response["blocks"] = self.blocks
+            elif return_nav:
+                response["navigation"] = self.navigation
```

Comment:
Yes, the world will not fall over.  :)
The response will only include the "root" field and not any of the other ones.  This will not happen in the normal case where the call comes through django's `reverse` or `urls.py`.